src/cpdb/helpers.py

```diff
@@ -22,31 +31,45 @@
 """a feedback from the group is to have a dictionary from the abbreviation for agency for something more explicity
 where would this list comes from? """
 
-
-def unpack_object(obj):
-    s = str(obj["Body"].read(), "utf8")
-    data = StringIO(s)
-    df = pd.read_csv(data, encoding="utf8")
-    return df
-
-
-def fetch_boto3_data(branch: str, table: str, previous=False):
-    if previous:
-        ver_key = f"db-cpdb/{branch}/2022-04-15/output/{table}.csv"
-    else:
-        ver_key = f"db-cpdb/{branch}/latest/output/{table}.csv"
-    client = boto3.client(
+def s3_resource():
+    return boto3.resource(
         "s3",
         aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
         aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
         endpoint_url=os.getenv("AWS_S3_ENDPOINT"),
     )
-    obj = client.get_object(
-        Bucket="edm-private",
-        Key=ver_key,
+
+def get_csv(bucket, csv_filename):
+    obj = s3_resource().Object(bucket_name=bucket, key=csv_filename)
+    s = str(obj.get()["Body"].read(), "utf8")
+    data = StringIO(s)
+    return pd.read_csv(data, encoding="utf8")
+
+def zip_from_DO(zip_filename, bucket):
+    zip_obj = s3_resource().Object(bucket_name=bucket, key=zip_filename)
+    buffer = BytesIO(zip_obj.get()["Body"].read())
+
+    return ZipFile(buffer)
+
+def unzip_shapefile(zipfile, table):
+    try:
+        with zipfile as zf:
+            if os.path.exists(f".library/{table}"):
```

Comment:
This doesn't completely solve the problem, as if someone else is mid zip-extraction, you could delete the folder out from under them.